pkg/operator/ceph/csi/controller_test.go

```diff
@@ -155,4 +155,86 @@ func TestCephCSIController(t *testing.T) {
 		assert.NoError(t, err)
 		assert.Equal(t, 2, len(ds.Items), ds)
 	})
+
+	t.Run("success ceph csi deployment with multus", func(t *testing.T) {
+		fakeClientSet := test.New(t, 1)
+		test.SetFakeKubernetesVersion(fakeClientSet, "v1.21.0")
+		c := &clusterd.Context{
+			Clientset:     fakeClientSet,
+			RookClientset: rookclient.NewSimpleClientset(),
+		}
+		_, err := c.Clientset.CoreV1().Pods(namespace).Create(ctx, test.FakeOperatorPod(namespace), metav1.CreateOptions{})
+		assert.NoError(t, err)
+		_, err = c.Clientset.AppsV1().ReplicaSets(namespace).Create(context.TODO(), test.FakeReplicaSet(namespace), metav1.CreateOptions{})
+		assert.NoError(t, err)
+		cephCluster := &cephv1.CephCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      namespace,
+				Namespace: namespace,
+			},
+			Spec: cephv1.ClusterSpec{
+				Network: cephv1.NetworkSpec{
+					Provider:  "multus",
+					Selectors: map[string]string{"public": "public-net", "cluster": "cluster-net"},
+				},
+			},
+			Status: cephv1.ClusterStatus{
+				Phase: "",
+				CephVersion: &cephv1.ClusterVersion{
+					Version: "14.2.9-0",
+				},
+				CephStatus: &cephv1.CephStatus{
+					FSID: "89d00089-d6e6-4445-b596-96898ce31793",
+				},
+			},
+		}
+		// Mock clusterInfo
+		secrets := map[string][]byte{
+			"fsid":         []byte(name),
+			"mon-secret":   []byte("monsecret"),
+			"admin-secret": []byte("adminsecret"),
+		}
+		secret := &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "rook-ceph-mon",
+				Namespace: namespace,
+			},
+			Data: secrets,
+			Type: k8sutil.RookType,
+		}
+		_, err = c.Clientset.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{})
+		assert.NoError(t, err)
+		s := scheme.Scheme
+		s.AddKnownTypes(cephv1.SchemeGroupVersion, &cephv1.CephCluster{}, &cephv1.CephClusterList{})
+
+		object := []runtime.Object{
+			cephCluster,
+		}
+		// Create a fake client to mock API calls.
+		cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(object...).Build()
+		c.Client = cl
+		// // Create a ReconcileCSI object with the scheme and fake client.
+		r := &ReconcileCSI{
+			client:  cl,
+			context: c,
+			opConfig: controller.OperatorConfig{
+				OperatorNamespace: namespace,
+				Image:             "rook",
+				ServiceAccount:    "foo",
+			},
+		}
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+
+		ds, err := c.Clientset.AppsV1().DaemonSets(namespace).List(ctx, metav1.ListOptions{})
+		assert.NoError(t, err)
+		assert.Equal(t, 4, len(ds.Items), ds)
+		assert.Equal(t, "csi-cephfsplugin", ds.Items[0].Name)
+		assert.Equal(t, "csi-cephfsplugin-holder-rook-ceph", ds.Items[1].Name)
+		assert.Equal(t, "csi-rbdplugin", ds.Items[2].Name)
+		assert.Equal(t, "csi-rbdplugin-holder-rook-ceph", ds.Items[3].Name)
```

Comment:
Will names remain in the same index always?

Reply:
Yes it's sorted.